bin/ad/ftw.c

```diff
@@ -293,13 +293,14 @@ open_dir_stream(int *dfdp, struct ftw_data *data, struct dir_data *dirp)
             DIR *st = data->dirstreams[data->actdir]->stream;
             struct dirent64 *d;
             size_t actsize = 0;
-            const size_t MAX_BUFFER_SIZE = 1024 * 1024; /* 1MB buffer limit */
+            const size_t MAX_ENTRY_SIZE = 255;
+            const size_t MAX_BUFFER_SIZE = 1024 * 1024;
 
             while ((d = __readdir64(st)) != NULL) {
                 size_t this_len = NAMLEN(d);
 
-                if (this_len > UTF8FILELEN_EARLY) {
-                    this_len = UTF8FILELEN_EARLY;
+                if (this_len > MAX_ENTRY_SIZE) {
```

Comment:
what's this doing?  I assume it's doing something sensible, I just don't know what UTF8FILELEN_EARLY means, or why it's being replaced

Reply:
Nope, not at all - the maximum length of something is inevitably going to be a magic number.  If it's intended to be the max file name length under OS X, though, I might suggest renaming the constant to be something like MAXFILENAMELEN or something, since 'MAX_ENTRY_SIZE' is a bit opaque in terms of semantics (this feels a bit like bikeshedding; if so, sorry!)

Reply:
This is a good point, and if nothing else it makes the code more readable to future readers which is always a good thing. Changeset updated!